handlers/middlewares/logPresence.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+const config = require('../../config');
+function getUsername(user) {
+	var str = "";
+	if (user.first_name) str += user.first_name;
+	if (user.last_name) str += " " + user.last_name;
+	if (user.username) str += " @" + user.username;
+	return str;
+}
+function log(ctx, next) {
+	if (!("presenceLog" in config.chats) || !config.chats.presenceLog || config.chats.presenceLog === null) return next();
```

Comment:
What if `config.chats` is `undefined`?
```suggestion
	if (!chats.presenceLog) return next();
```